### DISCUSSION TITLE
[Question]: How to configure session storage properly?

### DISCUSSION THREAD

**Discussion by cshenrik** [danny-avila/LibreChat#1840] [Q&A] [answered]
### What is your question?

When I enable social login and registration using OpenID, I notice the following warning in my logs upon startup:

Warning: connect.session() MemoryStore is not designed for a production environment, as it will leak memory, and will not scale past a single process.

My interpretation of this is that if I'm running multiple LibreChat containers behind a load balancer without sticky sessions, then users will not necessarily hit the container with their session, and will therefore get an error. Is that correct?

Redis seems to be the alternative to MemoryStore. However, the [LibreChat documentation](https://docs.librechat.ai/install/configuration/dotenv.html#other) says: "Redis support is experimental, you may encounter some problems when using it." 

So my question boils down to: How would I best run LibreChat with OpenID enabled, behind a load balancer that doesn't use sticky sessions?

### More Details

N/A

### What is the main subject of your question?

User System/OAuth

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment, marked as answer]
The main "problem" you encounter when enabling Redis is needing to manually flush your cache as needed.

If you can manage this, you should enable it for your use case so that sessions can be shared across instances.

It's marked as experimental because I haven't personally configured a setup like this yet. 

Also, right now the redis cache is also coupled with other values aside from sessions. This needs to be decoupled at some point as you wouldn't necessarily want to flush sessions but may want to flush values such as models fetched from OpenAI, etc.

**Reply by cshenrik** [reply]
Thank you, @danny-avila. I will give it a shot with Redis.

**Reply by danny-avila** [reply]
thanks for asking!
